operator/v1/types_ingress.go

```diff
@@ -282,6 +287,315 @@ var (
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// Modern is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+	// Custom is a security profile that allows for user-defined parameters.
+	SecurityProfileCustomType SecurityProfileType = "Custom"
+)
+
+// SecurityProfile defines the schema for a security profile.
+type SecurityProfile struct {
+	// ciphers is used to specify the cipher algorithms that are negotiated
+	// during the SSL/TLS handshake with an IngressController. Preface a cipher
+	// with a "!" to disable a specific cipher from being negotiated. For example,
+	// to use DES-CBC3-SHA but not EDH-DSS-DES-CBC3-SHA (yaml):
+	//
+	//   cipers:
+	//     - DES-CBC3-SHA
+	//     - !EDH-DSS-DES-CBC3-SHA
+	//
+	// If unset, the "Intermediate" Ciphersuites are used from here:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	Ciphers []*string `json:"ciphers,omitempty"`
+	// securityProtocol is used to specify one or more encryption protocols
+	// that are negotiated during the SSL/TLS handshake with the IngressController.
+	//
+	// If unset, the "Intermediate" Versions are used from here:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	//
+	// +optional
+	SecurityProtocol *SecurityProtocol `json:"securityProtocol,omitempty"`
+	// dhParamSize sets the maximum size of the Diffie-Hellman parameters used for generating
+	// the ephemeral/temporary Diffie-Hellman key in case of DHE key exchange. The final size
+	// will try to match the size of the server's RSA (or DSA) key (e.g, a 2048 bits temporary
+	// DH key for a 2048 bits RSA key), but will not exceed this maximum value. Only 1024 or 2048
+	// values are allowed.
+	//
+	// If unset, the "Intermediate" DH Parameter size is used from here:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	DHParamSize *DHParamSize `json:"dhParamSize,omitempty"`
+}
+
+// SecurityProtocol defines one or more security protocols used by
+// an IngressController to secure network connections.
+type SecurityProtocol struct {
+	// minimumVersion enforces use of SecurityProtocolVersion or newer on
+	// SSL connections initiated by an IngressController. minimumVersion
+	// must be lower than or equal to maximumVersion.
+	//
+	// If unset and maximumVersion is set, minimumVersion will be set
+	// to maximumVersion. If minimumVersion and maximumVersion are unset,
+	// the minimum version in "Intermediate" Versions are used from here:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	MinimumVersion *SecurityProtocolVersion `json:"minimumVersion,omitempty"`
```

Comment:
Will the latest generator show the possible SecurityProtocolVersion values from the alias type? If not, I think you have to describe every possible enum value and their meaning inline here (cc @sttts)

Reply:
I don't see the values from the alias type. I'll update the godoc.